### PR TITLE
🎉 Source Intercom: Added conversation_id field to conversation_part records

### LINKED ISSUE #9572: Source Intercom: conversion_parts is not fetched by the source
*Is this your first time deploying Airbyte*: No
*OS Version / Instance*: debian-10-buster, GCP e2-standard-2
*Deployment*: Docker
*Airbyte Version*: _0.32.0-alpha_
*Source/version*: Intercom 0.1.12
*Destination name/version*: BigQuery 0.5.1

I have two issues with Intercom:
• Tables on “Incremental append/deduped” Sync mode appear as empty in Bigquery but tables on Full refresh Overwrite have data in them
• On `conversation_parts` table with Full refresh sync mode, the field `conversation_id` is always null. This prevents me from linking them back to the full conversation entity
I tried using both normalization options and Access Token has all Read permissions enabled. Let me know if I can help with anything! I haven’t seen any issue related on Github

 https://airbytehq.slack.com/archives/C01MFR03D5W/p1642063449327900?thread_ts=1642063449.327900&cid=C01MFR03D5W

### PR DESCRIPTION
**Pull request by lgomezm**
## What
Records from the `ConversationParts` stream don't have `conversation_id`, so in a scenario where a connection is configured to sync `conversations` and `conversation_parts`, there is no hay to determine what `conversation_parts` belong to what `conversations` at the destination.

## How
Added the `conversation_id` property to the records yielded by the `ConversationParts` stream.

## Recommended reading order
1. `source.py`
2. `unit_test.py`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by lgomezm** [comment]
Since I added a new unit tests, here's the unit test execution:
Screen Shot 2022-03-16 at 2 00 57 PM
I had no success at running the integration tests. It looks like they are configured to run with a test account I don't have access to. Please run them when you get a chance.

File: airbyte-integrations/connectors/source-intercom/source_intercom/source.py [hunk 1]
```diff
@@ -300,6 +300,14 @@ class ConversationParts(ChildStreamMixin, IncrementalIntercomStream):
 
     def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
         return f"/conversations/{stream_slice['id']}"
+    
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        records = super().parse_response(response, stream_state, **kwargs)
+        conversation_id = response.json().get('id')
+        for conversation_part in records:
+            if 'conversation_id' not in conversation_part:
+                conversation_part['conversation_id'] = conversation_id
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-intercom/source_intercom/source.py:309]:
```suggestion
           conversation_part.setdefault("conversation_id", "conversation_id")
```

**Review by alafanechere — COMMENTED** [review]
Thank you for your contribution @lgomezm! I made a minor suggestion. I'll run the acceptance tests and will approve/request changes according to results 🥁

**Review by alafanechere — CHANGES_REQUESTED** [review]
@lgomezm the tests are failing because you did not run `./gradlew format`. Please run it and push the changes. You'll have to rename the unit test you declared to make the flake test pass. I'm ready to merge it afterward.

File: airbyte-integrations/connectors/source-intercom/source_intercom/source.py (hunk 1 above)
Comment (lgomezm) [reply airbyte-integrations/connectors/source-intercom/source_intercom/source.py]: Done in [c88184b](https://github.com/airbytehq/airbyte/pull/11206/commits/c88184b939d605be30fe5d7c3e0213bdfb3a48b9).

File: airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py [hunk 2]
```diff
@@ -76,3 +76,35 @@ def test_switch_to_standard_endpoint_if_scroll_expired(requests_mock):
         records += list(stream1.read_records(sync_mode=SyncMode, stream_slice=slice))
 
     assert stream1._endpoint_type == Companies.EndpointType.standard
+
+
+def test_switch_to_standard_endpoint_if_scroll_expired(requests_mock):
```
[the remark this answers is not in the document]
Comment (lgomezm) [reply airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py]: Good catch! Updated in [8aebb16](https://github.com/airbytehq/airbyte/pull/11206/commits/8aebb169b0a4b81facd347ba9d10e55f87b3cf8e)

**Comment by lgomezm** [comment]
@alafanechere I've addressed your comments and ran `gradlew format`. Please let me know if there's anything else I should do. Thanks for reviewing!

**Review by alafanechere — CHANGES_REQUESTED** [review]
@lgomezm I'm ready to publish the new version. **Could you please bump the version number in**:
* `airbyte-integrations/connectors/source-intercom/Dockerfile`
* `airbyte-config/init/src/main/resources/seed/source_definitions.yaml`

**Please also update the changelog in `docs/integrations/sources/intercom.md`.**

I'll then run the publish command from the comments. Afterward you need to locally run `./gradlew airbyte-config:init:processResources` . This will update the `/Users/augustin/workspace/airbyte/airbyte-config/init/src/main/resources/seed/source_specs.yaml`, please then commit this file.

I would have liked to do this myself but as I don't have push permission on your organization fork I need to ask you to do this 😄

**Comment by lgomezm** [comment]
@alafanechere In [70216ed](https://github.com/airbytehq/airbyte/pull/11206/commits/70216edb300ec9f0c51c93e08bd94fd50ede728b), I've updated:
* airbyte-integrations/connectors/source-intercom/Dockerfile
* airbyte-config/init/src/main/resources/seed/source_definitions.yaml
* docs/integrations/sources/intercom.md
Please let me know when you publish, so I can run that final gradle command and push the changes again.

**Comment by alafanechere** [comment]
@lgomezm the connector is published, please run `./gradlew airbyte-config:init:processResources` and commit + push the change made to the `source_specs.yaml` file. We can merge afterward 🎉

**Comment by lgomezm** [comment]
@alafanechere Done. `source_specs.yaml` was updated in [1dd4fd5](https://github.com/airbytehq/airbyte/pull/11206/commits/1dd4fd5b8eed2176323cebe01ba09b3d5a406ce7).

**Review by alafanechere — APPROVED** [review]
Thank you for this contrib @lgomezm !

**Pull request merged**